docs/source/tutorials_source/package/tutorial_pretrain_detectron2.py

```diff
@@ -0,0 +1,254 @@
+cfg = config.get_cfg()
+cfg.merge_from_file(cfg_path)
+
+# use cuda if possible
+cfg.MODEL.DEVICE = device
+
+# randomly initialize network
+cfg.MODEL.WEIGHTS = ""
+
+# detectron2 uses BGR by default but pytorch/torchvision use RGB
+cfg.INPUT.FORMAT = "RGB"
+
+# %%
+# Next, we can build the Detectron2 model and extract the ResNet50 backbone as
+# follows:
+
+detmodel = modeling.build_model(cfg)
+
+backbone = torch.nn.Sequential(
+    detmodel.backbone.bottom_up,
+    SelectFeatures('res5'),
+    torch.nn.AdaptiveAvgPool2d(1),
+).to(device)
+
+# %%
+# Finally, let's build SimCLR around the backbone as shown in the other
+# tutorials. For this, we only require an additional projection head.
+projection_head = lightly.models.modules.SimCLRProjectionHead(
+    input_dim=num_ftrs,
+    hidden_dim=num_ftrs,
+    output_dim=128,
+).to(device)
+
+# %%
+# Setup data augmentations and loaders
+# ------------------------------------
+#
+# We start by defining the augmentations which should be used for training.
+# We use the same ones as in the SimCLR paper but change the input size and
+# minimum scale of the random crop to adjust to our dataset. 
+#
+# We don't go into detail here about using the optimal augmentations.
+# You can learn more about the different augmentations and learned invariances
+# here: :ref:`lightly-advanced`.
+collate_fn = lightly.data.SimCLRCollateFunction(
+    input_size=input_size,
+    min_scale=0.85,
+)
+
+dataset_train_simclr = lightly.data.LightlyDataset(input_dir=data_path)
+
+dataloader_train_simclr = torch.utils.data.DataLoader(
+    dataset_train_simclr,
+    batch_size=batch_size,
+    shuffle=True,
+    collate_fn=collate_fn,
+    drop_last=True,
+    num_workers=num_workers
+)
+
+# %%
+# Self-supervised pre-training
+# -----------------------------
+# Now all we need to do is define a loss and optimizer and start training!
+
+criterion = lightly.loss.NTXentLoss()
+optimizer = torch.optim.SGD(
+    list(backbone.parameters()) + list(projection_head.parameters()),
+    lr=lr,
+    momentum=0.9
+)
+
+
+for e in range(max_epochs):
+
+    mean_loss = 0.
+    for (x0, x1), _, _ in dataloader_train_simclr:
+
+        x0 = x0.to(device)
+        x1 = x1.to(device)
+
+        y0 = projection_head(backbone(x0).flatten(start_dim=1))
+        y1 = projection_head(backbone(x1).flatten(start_dim=1))
+
+        # backpropagation
+        loss = criterion(y0, y1)
+        loss.backward()
+
+        optimizer.step()
+        optimizer.zero_grad()
+
+        # update average loss
+        mean_loss += loss.detach().cpu().item() / len(dataloader_train_simclr)
+
+    print(f'[Epoch {e:2d}] Mean Loss = {mean_loss:.2f}')
+
+
+# %%
+# Storing the checkpoint
+# -----------------------
+# Now, we can use the pre-trained backbone from the Detectron2 model. The code
+# below shows how to save it as a Detectron2 checkpoint called `my_model.pth`.
+detmodel.backbone.bottom_up = backbone[0]
```

Comment:
I would try to make it more clear that this line just uses the first module in backbone which is the original detectron2 backbone.